ui/shared/context/Notifications.jsx

```diff
@@ -0,0 +1,143 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+export const NotificationsContext = React.createContext({});
+
+export class Notifications extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      notifications: [],
+      storedNotifications: JSON.parse(localStorage.getItem('notifications') || '[]'),
+    };
+    this.value = {
+      ...this.state,
+      notify: this.notify,
+      removeNotification: this.removeNotification,
+      clearStoredNotifications: this.clearStoredNotifications,
+    };
+  }
+
+  componentDidMount() {
+    this.notify = this.notify.bind(this);
+    this.removeNotification = this.removeNotification.bind(this);
+    this.shift = this.shift.bind(this);
+    this.clearStoredNotifications = this.clearStoredNotifications.bind(this);
+
+    this.unlistenStorage = window.addEventListener('storage', (e) => {
+      if (e.key === 'notifications') {
+        this.setValue({
+          storedNotifications: JSON.parse(localStorage.getItem('notifications') || '[]'),
+        });
+      }
+    });
+
+    if (window.navigator.userAgent.indexOf('Firefox/52') !== -1) {
```

Comment:
Lets remove this ESR52 check since ESR60 has been out since 2018-05-09, and ESR52 became officially EOL on 2018-09-05.

Reply:
done